src/NuGet.Core/NuGet.Packaging/Licenses/LicenseExpressionTokenizer.cs

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace NuGet.Packaging.Licenses
+{
+    internal class LicenseExpressionTokenizer
+    {
+        private static Tuple<char, string> OpeningBracket = new Tuple<char, string>('(', "(");
+        private static Tuple<char, string> ClosingBracket = new Tuple<char, string>(')', ")");
+
+        private readonly string _value;
+        /// <summary>
+        /// A tokenizer for a license expression.
+        /// This implementation assumes that the input has been sanitized and that there are no invalid characters.
+        /// </summary>
+        /// <param name="value"></param>
+        internal LicenseExpressionTokenizer(string value)
+        {
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                throw new ArgumentNullException(nameof(value));
+            }
+            _value = value.Trim();
+        }
+
+        /// <summary>
+        /// The valid characters for a license identifier are a-zA-Z0-9.-+
+        /// The valid characters for a license expression are the above whitespace and ().
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        internal bool HasValidCharacters()
+        {
+            for (var i = 0; i < _value.Length; i++)
+            {
+                // If the character is not among these characters
+                if (!((_value[i] >= 'a' && _value[i] <= 'z') ||
+                    (_value[i] >= 'A' && _value[i] <= 'Z') ||
+                    (_value[i] >= '0' && _value[i] <= '9') ||
+                    _value[i] == ' ' ||
+                    _value[i] == '.' ||
+                    _value[i] == '-' ||
+                    _value[i] == '+' ||
+                    _value[i] == '(' ||
+                    _value[i] == ')'
+                    ))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Given a string, tokenizes by space into operators and values. The considered operators are, AND, OR, WITH, (, and ). 
+        /// </summary>
+        /// <returns>Tokens</returns>
+        internal IEnumerable<LicenseExpressionToken> Tokenize()
+        {
+            var potentialTokens = _value.Split(' ');
+            foreach (var token in potentialTokens)
```

Comment:
Should you use `StringSplitOptions.RremoveEmptyEntries` here?

Reply:
It won't throw as I have a length check (and tests) for it, but that's a great catch. 
Slipped my mind about how this split method worked.